scss/bitstyles/ui/forms.stories.mdx

```diff
@@ -0,0 +1,234 @@
+import { Canvas, Meta, Story } from '@storybook/addon-docs/blocks';
+
+<Meta title="UI/Data/Forms" />
+
+# Forms
+
+**See inputs & labels for examples of using inputs of various types.**
+
+When using form elements in a form, they should always be wrapped in a list element, as this indicates to the user that all the elements are closely related to each other. This also gives us the opportunity to apply some layout, such as adding space between elements.
+
+Form elements will take 100% of the available width, so you’ll likely want to constrain them within a grid or flex layout system to make that more readable and usable.
+
+Here are some examples using the grid utility classes to create common form layouts.
+
+## Login form
+
+<Canvas>
+  <Story name="Login form">
+    {`
+      <div class="a-content a-content--small">
+        <form class="a-card@m u-padding-xl@m" method="POST" action="">
+          <h3>Login</h3>
+          <ul class="a-list-reset u-margin-xl--bottom">
+            <li class="u-margin-m--bottom">
+              <label for="email">Email</label>
+              <input id="email" type="email" autocomplete="email" />
+            </li>
+            <li class="u-margin-m--bottom u-flex u-flex--col">
+              <label for="password">Password</label>
+              <input id="password" type="password" autocomplete="current-password" />
+              <div class="u-flex u-justify-end u-margin-xs--top">
+                <a href="/">Forgot password?</a>
+              </div>
+            </li>
+          </ul>
+          <ul class="a-list-reset u-flex u-items-center">
+            <li class="u-margin-xl--right">
+              <button type="submit" class="a-button u-h3">Login</button>
+            </li>
+            <li>
+              <a href="/">or sign up</a>
+            </li>
+          </ul>
+        </form>
+      </div>
+    `}
+  </Story>
+</Canvas>
+
+## Login form with errors
+
+When a user submits a form without fulfilling all the requirements (e.g. some required fields are empty, or fields have malformed data entered):
+
+- add the `aria-invalid="true"` attribute to the input
+- add a sibling element containing a description of what went wrong
+- Connect that sibling element to the input by adding an `aria-describedby="hint-element-id"` attribute to the input, and `id="hint-element-id"` (choose your own id!) to the hint’s element
```

Comment:
Not necessarily related to this PR, but what if I already have a hint text and am using the `aria-describedby` attribute there? Replace it with the error-feedback? Or is it fine to have two elements with this attribute?

Reply:
Which way around do you mean the multiple-ness is happening? :)

`aria-describedby` can be given a list of IDs if that element is described by multiple elements (e.g. an input that has a description on how to fill it out, like password requirements, and also has an error when it’s submitted without meeting them)

It’s totally fine to have multiple `aria-describedby` attributes on the page, and two elements can have that attirbute referring to the same ID if they are both described by the element with that ID

Does that answer your question? I’ll update the docs here with some of that! 🙇‍♂️

Reply:
> […] and two elements can have that attirbute referring to the same ID if they are both described by the element with that ID

That part answers my question, thanks 🙏